tests/unit/parallel/executors/shared_parallel_executor.cpp

```diff
@@ -32,6 +33,17 @@ struct shared_parallel_executor
     {
         return hpx::async(std::forward<F>(f));
     }
+
+    template <typename F>
+    typename std::remove_reference<
+        typename hpx::util::result_of<
+            typename hpx::util::decay<F>::type()
+        >::type
+    >::type
```

Comment:
I agree. Any suggestions on how to solve this? The patch here was just meant as a stop-gap measure to make tests pass. Any proper solution would be certainly preferred.